R/txrange.R

```diff
@@ -0,0 +1,265 @@
+#' Assigns each read class geneScore and txScore
+#' @param se summerized experiment object with read classes/ranges
+#' @param genomeSequence genomeSequence
+#' @param annotations GRangesList of annotations
+txrange.scoreReadClasses = function(se, genomeSequence, annotations, 
+                                    min.readCount = 2){
+    options(scipen = 999)
+    se = addRowData(se, genomeSequence, annotations)
+    thresholdIndex = which(rowData(se)$readCount
+        >=min.readCount)
+    rowData(se)$geneScore = getGeneScore(se, thresholdIndex)
+    rowData(se)$txScore = getTranscriptScore(se, thresholdIndex)
+    return(se)
+}
+
+#' calculates labels and features used in model generation
+addRowData = function(se, genomeSequence, annotations){
+  rowData(se)$numExons <- elementNROWS(rowRanges(se))
+  rowData(se)$equal = isReadClassEqual(rowRanges(se), annotations)
+  rowData(se)$compatible = isReadClassCompatible(rowRanges(se), annotations)
+  rowData(se)$GENEID = assignGeneIds(rowRanges(se), annotations)
+  rowData(se)$novel = grepl("gene.", 
+      rowData(se)$GENEID)
+  se = calculateGeneProportion(se)
```

Comment:
56f95a1 now I calculate geneReadCount at the same time as proportion but its memory inefficient as it stores the count multiple times since we don't have a structure for storing gene level information.